src/main/java/ro/developmentfactory/thedrones/service/DronePositionServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+
+    public DronePositionServiceImpl(DroneStatusRepository droneStatusRepository, DroneRepository droneRepository) {
+        this.droneStatusRepository = droneStatusRepository;
+        this.droneRepository = droneRepository;
+    }
+
+    @Override
+    @Transactional
+    public DroneStatusResponse turningDirection(DroneStatusResponse droneStatusResponse) {
+        UUID idDrone = droneStatusResponse.getIdDrone();
+
+        Drone drone = droneRepository.findById(idDrone)
+                .orElseThrow(() -> new EntityNotFoundException("Drone with id " + idDrone + " not found"));
+
+        if (droneStatusResponse.getTurnDirection() == null) {
+            throw new IllegalArgumentException("Turn direction cannot be null");
+        }
+
+        DroneStatus currentStatus = drone.getDroneStatus();
+
+        Direction newDirection;
+
+        if (droneStatusResponse.getTurnDirection() == DroneStatusResponse.TurnDirection.RIGHT) {
+            newDirection = currentStatus.getFacingDirection().turnRight();
+        } else {
+            newDirection = currentStatus.getFacingDirection().turnLeft();
+        }
+
+        currentStatus.setFacingDirection(newDirection);
+
+        droneStatusRepository.save(currentStatus);
+        return convertToResponse(currentStatus);
+    }
+
+    @Override
+    public DroneStatusResponse moveForward(DroneStatusResponse droneStatusResponse) {
+        UUID idDrone = droneStatusResponse.getIdDrone();
+        log.debug("Moving forward for drone: {}", idDrone);
+
+        Drone drone = droneRepository.findById(idDrone)
+                .orElseThrow(() -> new EntityNotFoundException("Drone with id " + idDrone + " not found"));
+
+        DroneStatus droneStatus = drone.getDroneStatus();
+        if (droneStatus == null) {
+            throw new IllegalArgumentException("DroneStatus must not be null");
+        }
+        validateMove(droneStatus);
+        updatePosition(droneStatus);
+
+        droneStatusRepository.save(droneStatus);
+        log.debug("Drone status updated successfully: {}", droneStatus);
+
+        return convertToResponse(droneStatus);
+    }
+
+    private void updatePosition(DroneStatus droneStatus) {
+        Position newPosition = calculateNewPosition(droneStatus);
+        int newX = newPosition.getX();
+        int newY = newPosition.getY();
+
+        if(newX >= 0 && newX < 10 && newY >= 0 && newY < 10) {
+            droneStatus.setCurrentPositionX(newX);
+            droneStatus.setCurrentPositionY(newY);
+            droneStatus.getDrone().setCountMove(droneStatus.getDrone().getCountMove() + 1);
+            log.debug("New position for drone:({}, {})", newX, newY);
+        }else {
+            log.error("Drone cannot move because it's a Drone cannot move because it's at the edge of the field. Current position: ({}, {}), New position: ({}, {})",
+                    droneStatus.getCurrentPositionX(), droneStatus.getCurrentPositionY(), newX, newY);
+        }
```

Comment:
you don't need to validate here as you were supposed to do this earlier. here you should only actually move forward as you have already met the prerequisite and you are fine to move